src/main/scala/dotty/dokka/tasty/BasicSupport.scala

```diff
@@ -17,6 +17,27 @@ trait BasicSupport:
     def topLevelEntryName(using ctx: Context): Option[String] = if (sym.isPackageDef) None else
       if (sym.owner.isPackageDef) Some(sym.name) else sym.owner.topLevelEntryName
 
+    def getVisibility(): ScalaVisibility = 
+      if (sym.flags.is(Flags.Private)) ScalaVisibility.Private
+      else if (sym.flags.is(Flags.Protected)) ScalaVisibility.Protected
+      else ScalaVisibility.NoModifier
+
+    def getModifier(): ScalaModifier = 
+      if (sym.flags.is(Flags.Case)) ScalaModifier.Case
+      else if (sym.flags.is(Flags.Abstract)) ScalaModifier.Abstract
+      else if (sym.flags.is(Flags.Sealed)) ScalaModifier.Sealed
+      else if (sym.flags.is(Flags.Final)) ScalaModifier.Final
+      else ScalaModifier.Empty
```

Comment:
This seems a bit more restrictive than it should be - a `case` class can be `abstract`, `sealed`, or `final`. An `abstract` class can be `sealed`, though it cannot be `final`. A `sealed` class cannot be `final`. Perhaps what makes the most sense is to simply return a list of `ScalaModifier`s?